src/components/Hoverable/index.tsx

```diff
@@ -0,0 +1,213 @@
+import _ from 'lodash';
+import React, {MutableRefObject, ReactElement, RefAttributes, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState} from 'react';
+import {DeviceEventEmitter} from 'react-native';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import CONST from '@src/CONST';
+import HoverableProps from './types';
+
+/**
+ * Maps the children of a Hoverable component to
+ * - a function that is called with the parameter
+ * - the child itself if it is the only child
+ * @param children The children to map.
+ * @param callbackParam The parameter to pass to the children function.
+ * @returns The mapped children.
+ */
+function mapChildren(children: ((isHovered: boolean) => ReactElement) | ReactElement | ReactElement[], callbackParam: boolean): ReactElement & RefAttributes<HTMLElement> {
+    if (Array.isArray(children) && children.length === 1) {
+        return children[0];
+    }
+
+    if (typeof children === 'function') {
+        return children(callbackParam);
+    }
+
+    return children as ReactElement;
+}
+
+/**
+ * Assigns a ref to an element, either by setting the current property of the ref object or by calling the ref function
+ * @param ref The ref object or function.
+ * @param element The element to assign the ref to.
+ */
+function assignRef(ref: MutableRefObject<HTMLElement> | ((element: HTMLElement) => void), element: HTMLElement) {
+    if (!ref) {
+        return;
+    }
+
+    if (typeof ref === 'function') {
+        ref(element);
+    } else if (_.has(ref, 'current')) {
+        // eslint-disable-next-line no-param-reassign
+        ref.current = element;
+    }
+}
+
+/**
+ * It is necessary to create a Hoverable component instead of relying solely on Pressable support for hover state,
+ * because nesting Pressables causes issues where the hovered state of the child cannot be easily propagated to the
+ * parent. https://github.com/necolas/react-native-web/issues/1875
+ */
+const Hoverable = React.forwardRef<HTMLElement, HoverableProps>(
```

Comment:
Improve it stylistically:
```ts
function Hoverable(
    {disabled = false, onHoverIn = () => {}, onHoverOut = () => {}, onMouseEnter = () => {}, onMouseLeave = () => {}, children, shouldHandleScroll = false}: HoverableProps,
    outerRef: ForwardedRef<HTMLElement>,
)
```

and move forwardRef to the export line:
```ts
export default forwardRef(Hoverable);
```

This is more accurate with TS guidelines